EXILED/Exiled.Events/Patches/Events/Player/ChangingRoleAndSpawned.cs

```diff
@@ -187,63 +192,54 @@ private static void ChangeInventory(ChangingRoleEventArgs ev)
         {
             try
             {
-                if (!NetworkServer.active || !ev.SpawnFlags.HasFlag(RoleSpawnFlags.AssignInventory))
-                {
+                if (ev.ShouldPreserveInventory || ev.Reason == API.Enums.SpawnReason.Destroyed)
                     return;
-                }
 
                 Inventory inventory = ev.Player.Inventory;
-                bool flag = InventoryItemProvider.KeepItemsAfterEscaping && ev.Reason == API.Enums.SpawnReason.Escaped;
-                if (flag)
+
+                if (ev.Reason == API.Enums.SpawnReason.Escaped)
                 {
-                    List<ItemPickupBase> list = new List<ItemPickupBase>();
+                    List<ItemPickupBase> list = new();
                     if (inventory.TryGetBodyArmor(out BodyArmor bodyArmor))
-                    {
                         bodyArmor.DontRemoveExcessOnDrop = true;
-                    }
 
-                    HashSet<ushort> hashSet = HashSetPool<ushort>.Pool.Get();
-                    foreach (KeyValuePair<ushort, ItemBase> item2 in inventory.UserInventory.Items)
+                    while (inventory.UserInventory.Items.Count > 0)
                     {
-                        if (item2.Value is Scp1344Item scp1344Item)
-                        {
-                            scp1344Item.Status = Scp1344Status.Idle;
-                        }
```

Comment:
also, are you sure that thing for 1344 is not necessary?

Reply:
Please note that you are looking at a conversion from a prefix patch to a transpiler. That code was copy&pasted from the base method for the prefix, it is not removed by the transpiler.

Reply:
Honestly, unsure, needs testing, but it didn't in the sl14 branch when I first started my changes.